src/hevm/hevm-cli/hevm-cli.hs

```diff
@@ -593,24 +596,36 @@ dappCoverage opts _ solcFile =
         let
           dapp = dappInfo "." contractMap sourceCache
           f (k, vs) = do
-            putStr ("\x1b[0m" ++ "————— hevm coverage for ") -- Prefixed with color reset
-            putStrLn (unpack k ++ " —————")
-            putStrLn ""
-            forM_ vs $ \(n, bs) -> do
-              case ByteString.find (\x -> x /= 0x9 && x /= 0x20 && x /= 0x7d) bs of
-                Nothing -> putStr "\x1b[38;5;240m" -- Gray (Coverage status isn't relevant)
-                Just _ ->
-                  case n of
-                    -1 -> putStr "\x1b[38;5;240m" -- Gray (Coverage status isn't relevant)
-                    0  -> putStr "\x1b[31m" -- Red (Uncovered)
-                    _  -> putStr "\x1b[32m" -- Green (Covered)
-              Char8.putStrLn bs
-            putStrLn ""
-
+            if shouldPrintCoverage (EVM.UnitTest.covMatch opts) k then do
```

Comment:
when shouldPrintCoverage
could also be cute here

Reply:
don't follow 😓

Reply:
Yup, `when x m = if x then m else pure ()`

Reply:
ohhh neat lemme give it a shot

Reply:
also, in general, hoogle is your friend. The nix shell also contains language server which I think may also have hoogle built in and you can get it from whatever editor youre using

Reply:
hmm ok how does this work when m is a multiline block like we have atm?

Reply:
Like this:
```
when x $
   do m
       ...
```

Reply:
And when you are done you just go back to previous indentation level

Reply:
aight just gave it a shot, in this case putting `do` on the same line as the `when` felt more natural, but lmk if you'd prefer on it a newline 😅

Reply:
sounds good too!